include/Options/Business/Sync.php

```diff
@@ -40,9 +40,9 @@ public static function key(): string {
 	 * @phpstan-return array{type: 'array', items: array{type: SchemaType, enum: non-empty-list<non-falsy-string>}}
 	 */
 	protected function get_data_structure(): array {
-		add_filter( 'lang_dir_for_domain', '__return_false' ); // Prevents loading the translations too early.
+		$GLOBALS['l10n_unloaded']['polylang'] = true;
 		$enum = array_keys( \PLL_Settings_Sync::list_metas_to_sync() );
-		remove_filter( 'lang_dir_for_domain', '__return_false' );
+		unset( $GLOBALS['l10n_unloaded']['polylang'] );
 		unload_textdomain( 'polylang', true ); // Required to allow `_load_textdomain_just_in_time()` to load the translations.
```

Comment:
```suggestion
```
I guess we don't need this line anymore. Do we ?

Reply:
We do.

Reply:
Our texts are not translated otherwise (the test also fails).

Reply:
I was surprised but understand now. This is not really the global `$l10n_unloaded` which prevents the translation but this [piece of code](https://github.com/WordPress/wordpress-develop/blob/6.7.2/src/wp-includes/l10n.php#L1412-L1417).

With this in mind, I would now propose:
```php
	protected function get_data_structure(): array {
		$GLOBALS['l10n']['polylang'] = new \NOOP_Translations();
		$enum = array_keys( \PLL_Settings_Sync::list_metas_to_sync() );
		unset( $GLOBALS['l10n']['polylang'] );
		return array(
			'type'  => 'array',
			'items' => array(
				'type' => $this->get_type(),
				'enum' => $enum,
			),
		);
	}
```
What do you think?